node/bindnode/infer_test.go

```diff
@@ -377,33 +404,48 @@ func TestPrototypePointerCombinations(t *testing.T) {
 				// Assigning with the missing field should only work with optional.
 				nb := proto.NewBuilder()
 				err = dagjson.Decode(nb, strings.NewReader(`{}`))
-				if modifier == "optional" {
+				switch modifier.schemaField {
+				case "optional", "optional nullable":
 					qt.Assert(t, err, qt.IsNil)
 					node := nb.Build()
 					// The resulting node should be non-nil with a nil field.
 					nodeVal := reflect.ValueOf(bindnode.Unwrap(node))
 					qt.Assert(t, nodeVal.Elem().FieldByName("Field").IsNil(), qt.IsTrue)
-				} else {
+				default:
 					qt.Assert(t, err, qt.Not(qt.IsNil))
 				}
 
 				// Assigning with a null field should only work with nullable.
 				nb = proto.NewBuilder()
 				err = dagjson.Decode(nb, strings.NewReader(`{"field":null}`))
-				if modifier == "nullable" {
+				switch modifier.schemaField {
+				case "nullable", "optional nullable":
 					qt.Assert(t, err, qt.IsNil)
 					node := nb.Build()
 					// The resulting node should be non-nil with a nil field.
 					nodeVal := reflect.ValueOf(bindnode.Unwrap(node))
-					qt.Assert(t, nodeVal.Elem().FieldByName("Field").IsNil(), qt.IsTrue)
-				} else {
+					if modifier.schemaField == "nullable" {
+						qt.Assert(t, nodeVal.Elem().FieldByName("Field").IsNil(), qt.IsTrue)
+					} else {
+						qt.Assert(t, nodeVal.Elem().FieldByName("Field").Elem().IsNil(), qt.IsTrue)
+					}
+				default:
 					qt.Assert(t, err, qt.Not(qt.IsNil))
 				}
 			})
 		}
 	}
 }
 
+func nilable(kind reflect.Kind) bool {
```

Comment:
do you need this, or can you look at the second return of `ptrOrNilable`?

Reply:
The funcs are unexported so they cannot be shared between the test and non-test packages - that's how I arrived at the smaller copy. Any other alternative is more lines of code AFAICT.